File: server/Services/ReportsService.cs


namespace helpReviews.Services;

public class ReportsService(ReportsRepository repo, RestaurantsService restaurantsService)
{
  private readonly ReportsRepository repo = repo;
  private readonly RestaurantsService restaurantsService = restaurantsService;

    internal Report CreateReport(Report reportData)
    {
      Restaurant restaurant = restaurantsService.GetRestaurantById(reportData.RestaurantId, reportData.CreatorId);
      // REVIEW maybe if we cared about the owner of the restaurant we could check that here too?
      Report report = repo.CreateReport(reportData);
      // NOTE if you don't like the count method of counting reports. Then the restaurants table could have a reportCount column, and you could increase that here, in the same way we did with visits.
      return report;
    }

    internal List<Report> GetRestaurantReports(int restaurantId, string userId)
    {
      Restaurant restaurant = restaurantsService.GetRestaurantById(restaurantId, userId);
      // I don't need the restaurant for anything else, since it makes all of it's checks
      List<Report> reports = repo.GetRestaurantReports(restaurantId);
      return reports;
    }
}

File: client/src/services/ReportsService.js
import {api} from './AxiosService.js'
import {logger} from '../utils/Logger.js'
import {Report}from '../models/Report.js'
import {AppState} from '../AppState.js'


class ReportsService{
  async getRestaurantReports(restaurantId){
    AppState.reports = []
    const response = await api.get(`api/restaurants/${restaurantId}/reports`)
    logger.log('📜', response.data)
    AppState.reports = response.data.map(report => new Report(report))
  }

  async createReport(reportData){
    const response = await api.post('api/reports', reportData)
    logger.log('📜✨', response.data)
    // REVIEW only push in if on the appropriate page
    if(AppState.activeRestaurant?.id == reportData.restaurantId){
      AppState.reports.push(new Report(response.data))
    }
  }
}

export const reportsService = new ReportsService();
